smack-bosh/src/main/java/org/jivesoftware/smack/bosh/XMPPBOSHConnection.java

```diff
@@ -197,6 +197,18 @@ protected void connectInternal() throws SmackException {
         saslFeatureReceived.checkIfSuccessOrWaitOrThrow();
 
         callConnectionConnectedListener();
+
+        // Automatically makes the login if the user was previously connected successfully
+        // to the server and the connection was terminated abruptly
+
+        // in cases for reconnect:
+        // if do this asynchronously in BOSHConnectionListener saslFeatureReceived.check
+        // method doesn't pass
+        // and login flow becomes not clear and complicated.
+        if (wasAuthenticated) {
```

Comment:
Newer Smack versions do not automatically login any more when `connect()` is invoked. See idiomatic way is chain the call, e.g. `connection.connect().login()`. So please remove this.